### DISCUSSION TITLE
selectOneMenu truncate long items

### DISCUSSION THREAD

**Discussion by Just-Dominic** [primefaces/community#753] [PrimeFaces] [answered]
Hi,

I'm struggling to get this right. 

I wan't to truncate the long options in my selectOneMenu. See the print-screen below.

I've changed the autoWidth property but I don't see any difference. The width is calculated in JS on click. So anything I do in CSS doesn't work. I just want to panel to be as long as the menu and truncate items that are too long. Am I missing something or... 

Anyone any ideas? 

![image](https://github.com/primefaces/community/assets/144908012/dd14be98-4399-409e-8490-b65202a58ed8)

**Comment by melloware** [comment]
Yep you want to change the width using panelClass or panelStyle and also have the CSS abbreviate to ellipses.

**Comment by Just-Dominic** [comment, marked as answer]
Hey Melloware,

Thanks for the answer. I had to use the appendTo attr. to append it to field itself (not to the body). This way the 100% width is correctly calculated and you can truncate long items.
